File: cpp/217_ContainsDuplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
    	std::unordered_map<int, int> hash;
    	for( size_t i = 0; i < nums.size(); ++i ){
    		if( hash.find(nums[i]) == hash.end() ){
    			hash[nums[i]] = 1;
    		} else {
    			return true;
    		}
    	}
    	return false;
    }
};

File: js/217_ContainsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
	var hash = {};
	for( var i in nums ){
		if( hash[nums[i]] === 1 ){
			return true;
		} else {
			hash[nums[i]] = 1;
		}
	}
	return false;
};